### PR TITLE
Download tool

### PR DESCRIPTION
**Pull request by alexellis**
Merge node.js download tool.

### CODE REVIEW

File: downloader/index.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+.then((urls)=> {
+    return new Promise((resolve, reject) => {
+        var tasks = [];
+        urls.forEach((url) => {
+            if(url.length > 0) {
+                let outfile = url.substring(url.lastIndexOf("\/\/")+2);
+                outfile = outfile.replace(/\//g, "_");
+                outfile = outfile.replace(/\./g, "_") + ".html";
+                tasks.push({ url: url, outfile: outfile });
+            }
```
Comment (alexellis) [inline downloader/index.js:29]: Sanitises the URL into a filename.

File: downloader/index.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      process.stdout.write(".")
+      request.get(task.url, (err,res,body) => {
+          fs.writeFile("html/" + task.outfile, body, "utf8", () => {
+              process.stdout.write("-")
+            // console.log("Written: " + task.outfile);
+            setImmediate(done);              
+          });
+      });
+  }, 10);
+  q.drain = () => {
```
Comment (alexellis) [inline downloader/index.js:46]: 10 requests at a time, could be more or less.. can be tuned.

**Comment by user24** [comment]
It's nice code, but not convinced this is an improvement on a one line bash script...

**Comment by alexellis** [comment]
What this offers is parallelism which wget does not. So rather than having to wait for each HTTP request to be made one at a time, it can launch a number at the same time.

:+1:  points:
- unit testable
- readable, adds no new software since we already assume Node is installed
- doesn't replace your wget script

Are you ok to merge it, since it is there alongside your script not actually replacing it totally?

**Comment by user24** [comment]
Ahhh yes that is a nice feature. Happy to merge :)

**Comment by alexellis** [comment]
Quick question.. what size is your html directory when done?

**Comment by user24** [comment]
The recipes html weighs in at 973.5 MB for me.

On Thu, May 19, 2016 at 8:09 PM, Alex Ellis notifications@github.com
wrote:

> Quick question.. what size is your html directory when done?
> 
> —
> You are receiving this because you modified the open/close state.
> Reply to this email directly or view it on GitHub
> https://github.com/user24/auntiesrecipes/pull/8#issuecomment-220421454

## 

Howard Yeend BA MSc | solidRed.co.uk  | @user24

**Pull request merged**